ddtrace/propagation/http.py

```diff
@@ -750,6 +752,9 @@ def _extract(headers):
                 else:
                     log.debug("no dd list member in tracestate from incoming request: %r", ts)
 
+        if in_gcp_function():
```

Comment:
directly invoked google cloud functions (that weren't invoked by some datadog instrumented service) will have a `tracecontext` header. This causes the function to not show up as a Trace Root Span (as it's parent_id is being set to the id specified in the tracecontext header which google uses for their own tracing solution.

Because of this, we want to drop the span_id context for cloud functions